src/components/Square.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react'
+
+//component
+const Square = ({
+  id,
+  currentTurn,
+  value,
+  disabled,
+  winningSquares,
+  handleClick,
+}: SquareProps) => {
+  const [stateValue, setValue] = useState<string>('')
+  const [isWinningSquare, setIsWinningSquare] = useState<boolean>(false)
+
+  /**
+   * function for checking if 'this' square is part of the set of
+   * winning squares, sets the state to true if it is
+   */
+  const checkIsWinningSquare = () => {
```

Comment:
Functions shouldn't rely on mutable external state for output. An input of a function should have the exact same output. I would separate this function out into it's own file in utils/ and accept it's dependencies as args. Only return true or false though, don't mutate the state in the function (that's an unexpected side-effect)